File: LeetCode/Python/Easy/2864-Maximum-Odd-Binary-Number/solution.py
class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        result=""

        n=len(s)

        ones=0

        for digit in s:
            if digit=='1':
                ones+=1

        digits=['0' for _ in range(0, n)]

        digits[n - 1]='1'

        ones-=1

        for index in range(0, ones):
            digits[index]='1'

        result=''.join(digits)

        return result

File: LeetCode/Rust/Easy/2864-Maximum-Odd-Binary-Number/solution.rs
impl Solution
{
    pub fn maximum_odd_binary_number(s: String) -> String
    {
        let mut result: String=String::from("");

        let n: usize=s.len();

        let mut ones: usize=s.chars().filter(|&c| c=='1').count();

        let mut digits: Vec<char>=vec!['0'; n];

        digits[n - 1]='1';

        ones-=1;

        for index in 0..ones
        {
            digits[index]='1';
        }

        result=digits.into_iter().collect::<String>();

        result
    }
}